File: src/main/java/arcana/common/items/AspectIcon.java
package arcana.common.items;

import arcana.common.aspects.Aspect;
import mcp.MethodsReturnNonnullByDefault;
import net.minecraft.item.Item;
import net.minecraft.util.ResourceLocation;

import javax.annotation.ParametersAreNonnullByDefault;

@ParametersAreNonnullByDefault
@MethodsReturnNonnullByDefault
public class AspectIcon extends Item {
	Aspect aspect;
	public AspectIcon(Aspect aspect){
		super(new Properties());
		setRegistryName(new ResourceLocation(aspect.id.getNamespace(), aspect.id.getPath()));
		this.aspect = aspect;
	}

	@Override
	public String getDescriptionId() {
		return aspect.translationKey;
	}
}

File: src/main/kotlin/arcana/common/items/aspect/AspectIcon.kt
package arcana.common.items.aspect

import arcana.common.aspects.Aspect
import net.minecraft.item.Item

class AspectIcon(var aspect: Aspect) : Item(Properties()) {
    init {
        registryName = aspect.id
    }
    override fun getDescriptionId() = aspect.translationKey
}